assets/js/frontend/single-product.js

```diff
@@ -34,16 +34,22 @@ jQuery( function( $ ) {
 
 			$tab.closest( 'li' ).addClass( 'active' );
 			$tabs_wrapper.find( $tab.attr( 'href' ) ).show();
-		})
-		.trigger( 'init' );
+		});
 
-	$( 'a.woocommerce-review-link' ).click( function() {
-		$( '.reviews_tab a' ).click();
+	$( 'body' )
+		.on( 'click', 'a.woocommerce-review-link', function() {
+			$( '.reviews_tab a' ).click();
 		return true;
 	});
 
 	// Star ratings for comments
-	$( '#rating' ).hide().before( '<p class="stars"><span><a class="star-1" href="#">1</a><a class="star-2" href="#">2</a><a class="star-3" href="#">3</a><a class="star-4" href="#">4</a><a class="star-5" href="#">5</a></span></p>' );
+	$( 'body' )
+		.on( 'init', '#rating', function() {
+			$( '#rating' ).hide().before( '<p class="stars"><span><a class="star-1" href="#">1</a><a class="star-2" href="#">2</a><a class="star-3" href="#">3</a><a class="star-4" href="#">4</a><a class="star-5" href="#">5</a></span></p>' );
+		});
+	
+	//Init Tabs and Rating	
+	$( '.wc-tabs-wrapper, .woocommerce-tabs, #rating' ).trigger('init');
 
 	$( 'body' )
 		.on( 'click', '#respond p.stars a', function() {
```

Comment:
If we choose to group the above additional `$( ' body' )` into the first as suggested by @claudiosmweb , then it will be better to group to this altogether as well.